lms/djangoapps/courseware/views/views.py

```diff
@@ -167,7 +167,8 @@ def courses(request):
 
     # Do not add programs to the context if there are no program types enabled for the site.
     if program_types:
-        programs_list = get_programs_with_type(program_types)
+        programs_list = [program for program in get_programs_with_type(program_types)
```

Comment:
Move the filtering to `get_programs_with_type()` so that other consumers can take advantage of it. Add a kwarg, `include_hidden`, that defaults to `True`.